### PR TITLE
✅ (Unit Tests, UI, Bug Fixes, Enhancements) : Unit Testing, Mobile Scaling, Job Summary Enhancements

### PR DESCRIPTION
**Pull request by PxttytheThxtty**

### CODE REVIEW

File: libs/au-pair-dashboard/feature/src/lib/au-pair-dashboard.component.spec.ts [hunk 1]
```diff
@@ -147,3 +154,54 @@ describe('UserReportModalComponent', () => {
     expect(await component.closeModal).toReturn();
   });
 });
+
+describe('ParentRatingModalComponent', () => {
+  let component: ParentRatingModalComponent;
+  let fixture: ComponentFixture<ParentRatingModalComponent>;
+  let store: Store;
+
+  const validForm = {rating: [5]}
```
Comment (Ruben-Brits) [inline libs/au-pair-dashboard/feature/src/lib/au-pair-dashboard.component.spec.ts:163]: unused import

File: libs/au-pair-dashboard/feature/src/lib/au-pair-dashboard.component.spec.ts [hunk 2]
```diff
@@ -147,3 +154,54 @@ describe('UserReportModalComponent', () => {
...
+      imports: [IonicModule, CommonModule,HttpClientTestingModule,NavbarModule, RouterTestingModule, FormsModule,NgxsModule.forRoot([AppState])],
+      providers: [
+      {
+        provide:API, useValue:apiMock
+      }, 
+      ToastController, 
+      ModalController]
+    }).compileComponents();
+
+    store = TestBed.inject(Store);
```
Comment (Ruben-Brits) [inline libs/au-pair-dashboard/feature/src/lib/au-pair-dashboard.component.spec.ts:177]: unused import

File: libs/au-pair-dashboard/feature/src/lib/parent-rating-modal/parent-rating-modal.component.ts [hunk 3]
```diff
@@ -10,9 +10,8 @@ import { Store } from '@ngxs/store';
   styleUrls: ['./parent-rating-modal.component.scss'],
 })
 export class ParentRatingModalComponent implements OnInit {
-  parentID = "";
+  parentID = "";  
   auPairID = "";
-  public navParams = new NavParams;
```
Comment (Ruben-Brits) [inline libs/au-pair-dashboard/feature/src/lib/parent-rating-modal/parent-rating-modal.component.ts:15]: NavParams imported but not used

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.ts [hunk 4]
```diff
@@ -225,87 +230,61 @@ export class EditParentProfileComponent implements OnInit{
...
     if(val.medicalAidMM === "")
     {
-      emptyInput = true;
-      if(dom != null)
-      {
-        dom.innerHTML = "Main Member Name is empty";
-        dom.style.display = "block";
-      }
-    }else
+      console.log("Here 1");
```
Comment (Ruben-Brits) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.ts:233]: console log

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.ts [hunk 5]
```diff
@@ -225,87 +230,61 @@ export class EditParentProfileComponent implements OnInit{
...
     if(val.medicalAidMS === "")
     {
-      emptyInput = true;
-      if(dom != null)
-      {
-        dom.innerHTML = "Main Member Surname is empty";
-        dom.style.display = "block";
-      }
-    }else
+      console.log("Here 2");
```
Comment (Ruben-Brits) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.ts:243]: console log

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.ts [hunk 6]
```diff
@@ -225,87 +230,61 @@ export class EditParentProfileComponent implements OnInit{
...
     if(val.medicalAidNo === "")
     {
-      emptyInput = true;
-      if(dom != null)
-      {
-        dom.innerHTML = "Medical Aid Number is empty";
-        dom.style.display = "block";
-      }
-    }else
+      console.log("Here 3");
```
Comment (Ruben-Brits) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.ts:253]: console log

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.ts [hunk 7]
```diff
@@ -225,87 +230,61 @@ export class EditParentProfileComponent implements OnInit{
...
     if(val.medicalAidProvider === "")
     {
-      emptyInput = true;
-      if(dom != null)
-      {
-        dom.innerHTML = "Medical Aid Provider is empty";
-        dom.style.display = "block";
-      }
-    }else
+      console.log("Here 4");
```
Comment (Ruben-Brits) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.ts:263]: console log

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.ts [hunk 8]
```diff
@@ -225,87 +230,61 @@ export class EditParentProfileComponent implements OnInit{
...
     if(val.medicalAidPlan === "")
     {
-      emptyInput = true;
-      if(dom != null)
-      {
-        dom.innerHTML = "Medical Aid Plan is empty";
-        dom.style.display = "block";
-      }
-    }else
+      console.log("Here 5");
```
Comment (Ruben-Brits) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.ts:273]: console log

File: libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.ts [hunk 9]
```diff
@@ -371,17 +350,17 @@ export class EditParentProfileComponent implements OnInit{
...
-    .toPromise()
-    .then(
-      res => {
-        console.log("The response is:" + res); 
-      },
-      error=>{
-        console.log("Error has occured with API: " + error);
-      }
-    )
+    // this.parent.medID = medAid.mID;
```
Comment (Ruben-Brits) [inline libs/edit-parent-profile/feature/src/lib/edit-parent-profile.component.ts:353]: Can we remove or will this be used in future?

File: libs/explore/feature/src/lib/explore.component.spec.ts [hunk 10]
```diff
@@ -23,8 +36,14 @@ describe('ExploreComponent', () => {
...
        ],
-       providers:[API]
+       providers:[
+        {
+          provide:API, useValue:apiMock
+        }
+      ]
     }).compileComponents();
+
+    store = TestBed.inject(Store);
```
Comment (Ruben-Brits) [inline libs/explore/feature/src/lib/explore.component.spec.ts:46]: store imported but not used

**Review by Ruben-Brits — CHANGES_REQUESTED** [review]

**Review by Ruben-Brits — APPROVED** [review]

**Review by ricecakesreece — APPROVED** [review]

**Pull request merged**